packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/listen/assertListenRateLimiter.test.ts

```diff
@@ -1,178 +1,272 @@
-import { beforeEach, describe, expect, it } from "vitest";
-import { LISTENS_RATE_LIMIT_IP_PREFIX, LISTENS_RATE_LIMIT_TRACK_PREFIX, config } from "../../config";
-import { getRedisConnection } from "../../redis";
-import { listenRouteRateLimiter, listensIpRateLimiter, listensIpTrackRateLimiter } from "./listen";
+import { beforeEach, describe, expect, it } from 'vitest'
+
+import {
+  LISTENS_RATE_LIMIT_IP_PREFIX,
+  LISTENS_RATE_LIMIT_TRACK_PREFIX,
+  config
+} from '../../config'
+import { getRedisConnection } from '../../redis'
+
+import {
+  listenRouteRateLimiter,
+  listensIpRateLimiter,
+  listensIpTrackRateLimiter
+} from './listen'
 
 beforeEach(async () => {
-    // run with npm run test with audius-compose up
-    config.redisUrl = "redis://0.0.0.0:6379/00"
-    const redis = await getRedisConnection()
-    let cursor = '0';
-    do {
-        for await (const key of redis.scanIterator()) {
-            if (key.startsWith(LISTENS_RATE_LIMIT_IP_PREFIX) || key.startsWith(LISTENS_RATE_LIMIT_TRACK_PREFIX)) {
-                await redis.del(key)
-            }
-        }
-    } while (cursor !== '0')
+  // run with npm run test with audius-compose up
+  config.redisUrl = 'redis://audius-protocol-discovery-provider-redis-1/00'
```

Comment:
oh nvm my earlier comment, this should work

Reply:
Yeah we basically have to run this inside container only - see description

Reply:
I tried for a bit to explore other options, but I don't think there's a good way yet to support exposing these services to the host. I poked around at different solutions we could maybe use nginx for but figured this was easiest to unblock for now. I wonder if maybe we should just mock redis for these tests anyway